torch/nested/_internal/ops.py

```diff
@@ -380,8 +381,29 @@ def prod_dim_int(func, *args, **kwargs):
     return NestedTensor(func(inp._values, **new_kwargs), **extract_kwargs(args[0]))
 
 
+# Ordinarily, if a program calls .tolist compiling still works because there is
+# special handling in dynamo, but for tensor subclasses if .tolist is called
+# inside torch dispatch, the .tolist call will be directly on a FakeTensor.
+# This would result in an error since wrapper subclasses don't have storage.
+# To avoid this, we handle the fake tensor case in a special way.
+def sym_tolist(x):
+    if is_fake(x):
+        assert isinstance(x.shape[0], torch.SymInt)
```

Comment:
use `free_symbols` util, `from torch.fx.experimental.symbolic_shapes import free_symbols` - instead of peering into the shape at index 0.

Reply:
Oh cool, thanks. I ended up not using it here since I know that its a 1D tensor (added an assertion), but found a different place where this seemed useful.